### PR TITLE
Reworks some TETH abnormalities

### PR DESCRIPTION
**Pull request by Coxswain-Navigator**
## About The Pull Request
Improves some abnormalities that I feel are a bit incomplete

- Fairy Long Legs now has visual effects for the rain instead of making players use their imagination. It was fine otherwise.
- Fairy Gentleman now has a breach, and is a bit more difficult to squeeze for infinite easy sanity healing. In addition, I rewrote all of the dialogue because they didn't seem to be written by a native speaker. I may have gone a bit overboard when trying to make the dialogue seem interesting.
- So That No One Will Cry now has a breach and mechanics that lead to a breach as well.
- SO That No One Will Cry work damaged changed from WHITE to BLACK
- Changes some TETH gifts from HATs to HELMETs. I ran the numbers and there were 10+ "hat" gifts and NO helmet gifts whatsoever. Also gives a helmet gift to so that no one will cry.

Thanks to Mel Taculo for making the sprites on such short notice.

## Why It's Good For The Game
More variety for our early abnormalities

## Changelog
:cl:
tweak: tweaked some TETH abnormalities
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/abnormality/teth/fairy_gentleman.dm [hunk 1]
```diff
@@ -22,36 +36,163 @@
...
+		return
+
+	var/dist = get_dist(target, src)
+	if(jump_cooldown <= world.time)
+		var/chance_to_jump = 25
+		if(dist > 3)
+			chance_to_jump = 100
+		if(prob(chance_to_jump))
+			FairyJump(target)
+			return
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/teth/fairy_gentleman.dm:153]:
```suggestion
```

**Review by LanceSmites328 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/abnormality/teth/fairy_gentleman.dm [hunk 2]
```diff
@@ -22,36 +37,163 @@
...
+	var/list/angry = list(
+				"I'll wring you out!",
+				"Come on, I'm taking you for a ride!",
+				"This is all I got!",
+				"I'll be havin' this!",
+				"Scram!"
 				)
 
+//Action Buttons
+	attack_action_types = list(/datum/action/innate/abnormality_attack/toggle/FairyJump_Toggle)
```
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/teth/fairy_gentleman.dm:73]: Against the naming conventions.

File: code/modules/mob/living/simple_animal/abnormality/teth/fairy_gentleman.dm [hunk 3]
```diff
@@ -22,36 +36,163 @@
...
+		FairyJump(target)
+		return
+	if(!ishuman(target))
+		return ..()
+	var/mob/living/carbon/human/H = target
+	H.drunkenness += 5
+	to_chat(H, "<span class='warning'>Yuck, some of it got in your mouth!</span>")
+	if(H.sanity_lost)
+		melee_damage_type = RED_DAMAGE
+		return ..()
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/teth/fairy_gentleman.dm:129]:
```suggestion
```

File: code/modules/mob/living/simple_animal/abnormality/teth/so_that_no_cry.dm [hunk 4]
```diff
@@ -50,12 +74,154 @@
...
+	can_act = FALSE
+	damage_reflection = TRUE
+	damage_taken = 0
+	playsound(src, 'sound/abnormalities/so_that_no_cry/prepare.ogg', 50, TRUE, 7)
+	visible_message("<span class='warning>[src] assumes a stance!</span>")
+	icon_state = "so_that_no_cry_guard"
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 0, WHITE_DAMAGE = 0, BLACK_DAMAGE = 0, PALE_DAMAGE = 0)
+	SLEEP_CHECK_DEATH(10 SECONDS)
+	icon_state = icon_living
+	damage_coeff = list(BRUTE = 1, RED_DAMAGE = 1.5, WHITE_DAMAGE = 1, BLACK_DAMAGE = 1, PALE_DAMAGE = 2)
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/teth/so_that_no_cry.dm:176]: This needs to be updated to reflect #1326 . In this case it'd be
```suggestion
	damage_coeff.ChangeResistances(list(RED_DAMAGE = 0, WHITE_DAMAGE = 0, BLACK_DAMAGE = 0, PALE_DAMAGE = 0))
	SLEEP_CHECK_DEATH(10 SECONDS)
	icon_state = icon_living
	damage_coeff.ChangeResistances(list(RED_DAMAGE = 1.5, WHITE_DAMAGE = 1, BLACK_DAMAGE = 1, PALE_DAMAGE = 2))
```
This *may* need to be rebased for this to properly apply.

**Review by LanceSmites328 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/abnormality/teth/so_that_no_cry.dm [hunk 5]
```diff
@@ -50,12 +74,154 @@
...
+	can_act = FALSE
+	damage_reflection = TRUE
+	damage_taken = 0
+	playsound(src, 'sound/abnormalities/so_that_no_cry/prepare.ogg', 50, TRUE, 7)
+	visible_message("<span class='warning>[src] assumes a stance!</span>")
+	icon_state = "so_that_no_cry_guard"
+	ChangeResistances(list(BRUTE = 1, RED_DAMAGE = 0, WHITE_DAMAGE = 0, BLACK_DAMAGE = 0, PALE_DAMAGE = 0))
+	SLEEP_CHECK_DEATH(10 SECONDS)
+	icon_state = icon_living
+	ChangeResistances(list(BRUTE = 1, RED_DAMAGE = 1.5, WHITE_DAMAGE = 1, BLACK_DAMAGE = 1, PALE_DAMAGE = 2))
```
Comment (LanceSmites328) [inline code/modules/mob/living/simple_animal/abnormality/teth/so_that_no_cry.dm:176]: Heads up, the Brute change here and in the previous one is redundant, as this proc only changes the provided resistances. Not a real issue but I should let you know.

**Review by LanceSmites328 — APPROVED** [review]

**Pull request merged**